clients/src/main/java/org/apache/kafka/common/requests/UpdateMetadataRequest.java

```diff
@@ -117,20 +123,20 @@ public UpdateMetadataRequest(int controllerId, int controllerEpoch, Set<BrokerEn
         for (BrokerEndPoint brokerEndPoint : brokerEndPoints) {
             Map<SecurityProtocol, EndPoint> endPoints = Collections.singletonMap(SecurityProtocol.PLAINTEXT,
                     new EndPoint(brokerEndPoint.host(), brokerEndPoint.port()));
-            brokers.add(new Broker(brokerEndPoint.id(), endPoints));
+            brokers.add(new Broker(brokerEndPoint.id(), endPoints, null));
         }
         return brokers;
     }
 
     /**
-     * Constructor for version 1.
+     * Constructor for version 2.
      */
     public UpdateMetadataRequest(int controllerId, int controllerEpoch, Map<TopicPartition,
             PartitionState> partitionStates, Set<Broker> liveBrokers) {
-        this(1, controllerId, controllerEpoch, partitionStates, liveBrokers);
+        this(2, controllerId, controllerEpoch, partitionStates, liveBrokers);
     }
 
-    private UpdateMetadataRequest(int version, int controllerId, int controllerEpoch, Map<TopicPartition,
+    public UpdateMetadataRequest(int version, int controllerId, int controllerEpoch, Map<TopicPartition,
```

Comment:
@allenxwang Actually it's probably fine as it is since we would raise an error in `ProtoUtils.requestSchema()`. I didn't notice that this also supports version 0, so would it make sense change ControllerChannelManager to use this constructor for all cases. (And apologies for all this nitpicking)